learn/getting-started/writing-a-grpc-service-with-ballerina.md

```diff
@@ -0,0 +1,238 @@
+        - [Update and run the application](#update-and-run-the-application)
+
+## Prerequisites
+To complete this tutorial, you need:
+
+- A command terminal
+- A text editor. [VSCode](https://code.visualstudio.com/) with the [Ballerina extension](https://marketplace.visualstudio.com/items?itemName=WSO2.ballerina) installed is our preferred 
+  choice, and it has good support for Ballerina.
+- A Ballerina installation. Follow the steps in [Installing Ballerina](https://ballerina.io/learn/installing-ballerina/setting-up-ballerina/).
+- Some experience in gRPC service at least one programming language
+
+## Creating Service Definition
+
+In an RPC program, the first step is to define the interface of the service using an Interface Definition Language(IDL). gRPC commonly uses Protocol Buffers as IDL.
+Here we are going to use the simple service definition to illustrate the server and client implementation in Ballerina Language.
+
+```proto
+syntax = "proto3";
+
+package helloworld;
+
+// The greeting service definition.
+service Greeter {
+ // Sends a greeting
+ rpc sayHello (HelloRequest) returns (HelloReply);
+}
+
+// The request message with the user's name.
+message HelloRequest {
+ string name = 1;
+}
+
+// The response message with the greetings
+message HelloReply {
+ string message = 1;
+}
+```
+> Note: The sample service definition is taken from the quick start guide in gRPC official site.
+
+Let’s start implementing Ballerina service and client for the above definition.
+
+## Implementation
+
+As illustrated in the diagram below, we first need to compile the service definition file(helloworld.proto) and generate the source code for both client and service applications. In Ballerina, we can generate the source code using the built-in 'Protocol Buffers to Ballerina' tool.
+
+![gRPC Getting Started](/learn/images/grpc-getting-started.png)
+
+Let’s walk through implementing gRPC service and client in Ballerina Language.
+
+### Developing a gRPC service
+
+As with any other Ballerina Program, we need to start with creating a Ballerina project. Then we can generate service code in the project and write business logic.
+
+#### Create a Ballerina project
+
+A new Ballerina project can be created by executing the following command in the directory in which you want to create the project.
+
+```$ bal new greeter_service```
+
+> Note: For more information on creating Ballerina packages, please see the link.
+
+Automatically created `main.bal` file along with project creation is removed, As we are not going to use it in this guide.
+
+#### Generate source code
+
+The service code for the service definition can be generated by executing the following command inside the Ballerina project directory.
+
+```$ bal grpc --mode service --input helloworld.proto --output greeter_service/```
+
+Once successfully executed, you will see the below output.
+
+```
+Successfully extracted library files.
+Successfully generated ballerina file.
```

Comment:
```suggestion
Successfully generated the Ballerina file.
```

Reply:
This is the output generated from the tool. we need to change the tool as well